src/csharp/Microsoft.Spark/ML/Feature/Word2VecModel.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Sql;
+
+namespace Microsoft.Spark.ML.Feature
+{
+    public class Word2VecModel : IJvmObjectReferenceProvider
+    {
+        private static readonly string s_word2VecModelClassName = 
+            "org.apache.spark.ml.feature.Word2VecModel";
+        
+        private readonly JvmObjectReference _jvmObject;
+
+        /// <summary>
+        /// Create a <see cref="Word2VecModel"/> without any parameters
+        /// </summary>
+        public Word2VecModel()
+        {
+            _jvmObject = SparkEnvironment.JvmBridge.CallConstructor(s_word2VecModelClassName);
+        }
+
+        /// <summary>
+        /// Create a <see cref="Word2VecModel"/> with a UID that is used to give the
+        /// <see cref="Word2VecModel"/> a unique ID
+        /// </summary>
+        /// <param name="uid">An immutable unique ID for the object and its derivatives.</param>
+        public Word2VecModel(string uid)
+        {
+            _jvmObject = SparkEnvironment.JvmBridge.CallConstructor(s_word2VecModelClassName, uid);
+        }
+        
+        internal Word2VecModel(JvmObjectReference jvmObject)
+        {
+            _jvmObject = jvmObject;
+        }
+        
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Transform a sentence column to a vector column to represent the whole sentence.
+        /// </summary>
+        /// <param name="documentDF"><see cref="Word2VecModel"/> to transform</param>
+        public DataFrame Transform(DataFrame documentDF) => 
+            new DataFrame((JvmObjectReference)_jvmObject.Invoke("transform", documentDF));
+        
+        /// <summary>
+        /// Find "num" number of words whose vector representation most similar to the supplied
```

Comment:
You'll have to fix the max char line after this change. 
```suggestion
        /// Find <paramref name="num"/> number of words whose vector representation most similar to the supplied
```